src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -266,26 +272,33 @@ internal SourceText TryReadFileContent(CommandLineSourceFile file, IList<Diagnos
         /// <param name="messageProvider">Used to produce diagnostics.</param>
         /// <param name="diagnostics">Any produced diagnostics are added to this bag.</param>
         /// <returns>
-        /// A list of diagnostic options, where the dictionary index corresponds
+        /// A list of diagnostic and analyzer options, where the dictionary index corresponds
         /// to the options for the source path at the same index. If there are
-        /// no diagnostic options for the given path, the entry is null. Otherwise,
-        /// the options are a map from diagnostic ID to diagnostic severity, and
+        /// no options for the given path, the entry is null. Otherwise, the diagnostic
+        /// options are a map from diagnostic ID to diagnostic severity, and
         /// correspond to options provided for <see cref="SyntaxTree.DiagnosticOptions"/>.
+        /// The analyzerOptions are an OptionSet consisting of all properties that were not
+        /// diagnostic options.
         /// </returns>
-        internal static ImmutableArray<ImmutableDictionary<string, ReportDiagnostic>> GetAnalyzerConfigOptions(
-            IReadOnlyList<string> sourcePaths,
-            ArrayBuilder<EditorConfig> analyzerConfigs,
-            CommonMessageProvider messageProvider,
-            DiagnosticBag diagnostics)
+        internal static (ImmutableArray<ImmutableDictionary<string, ReportDiagnostic>> treeOptions,
+                         ImmutableArray<ImmutableDictionary<string, string>> analyzerOptions)
+            GetAnalyzerConfigOptions(IReadOnlyList<string> sourcePaths,
+                                     IReadOnlyList<EditorConfig> analyzerConfigs,
+                                     CommonMessageProvider messageProvider,
+                                     DiagnosticBag diagnostics)
```

Comment:
i had a lot of trouble understanding what this was even parsed as.  perhaps consider a simple named struct for the return type of this method.